workspaces/adventure-pack/goodies/typescript/Iterator.from/index.ts

```diff
@@ -0,0 +1,43 @@
+import "../Iterator.prototype.toIterable";
+
+declare global {
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  class Iterator<T> {
+    static from<TElem>(
+      object:
+        | Iterator<TElem>
+        | Iterable<TElem>
+        | { next(): IteratorResult<TElem> },
+    ): IterableIterator<TElem>;
+  }
+}
+
+(globalThis as Record<string, unknown>).Iterator ??= {};
```

Comment:
Btw I think a nice test for `Iterator.from` would be if we can replace some of the logic in `Iterator.prototype.flatMap` with this function.

Reply:
Nice I can create an issue for this in another PR!